src/PrestaShopBundle/Form/Admin/Product/ProductCombination.php

```diff
@@ -87,6 +83,7 @@ public function __construct($translator, $legacyContext)
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $currency = $this->legacyContext->getContext()->currency->iso_code;
```

Comment:
to avoid confusing it with Currency entity.
```suggestion
        $currencyIsoCode = $this->legacyContext->getContext()->currency->iso_code;
```
Btw it is old debate, but I remember Pablo always requested to only inject values that are used. In this case, instead of legacyContext, we could inject only the currency Iso code. I know your argument is smth like "container service definition is cleaner this way", but I still think less is better (and it would be easier to unit test - providing iso_code instead of mocking whole legacy context).

Reply:
You are right about the `LegacyContext` injection, because this is a poorly written service. But I did not wanted to refactor this further as it is out of the scope of this PR.


I think this is not a good idea to inject scalar values from the DI because

- We can't use autowiring
- We have many calls like this one in the same service definition.
- Calls are made even if we don't need them.
- We can't track function calls in the container, which hardens refactoring when we try to find usages.
- PS is the only project I've seen in all the projects i've been working on making this, which makes me think that it is either a genius idea or a super bad idea. Okay this is absolutely subjective, but you got the point :D

I'm not even talking about the poor developer experience, because sometimes we inject only a scalar value, sometimes we want more fields, and we inject many times the same field.

> and it would be easier to unit test

We don't have any unit test on any form so I don't think this is a valid argument on this case 😅.

Reply:
Created an adr issue for this : https://github.com/PrestaShop/ADR/issues/30